server/fleet/app.go

```diff
@@ -428,11 +428,11 @@ func (c *AppConfig) Copy() *AppConfig {
 
 	// HostExpirySettings: nothing needs cloning
 
-	if c.Features.AdditionalQueries != nil {
-		aq := make(json.RawMessage, len(*c.Features.AdditionalQueries))
-		copy(aq, *c.Features.AdditionalQueries)
-		c.Features.AdditionalQueries = &aq
+	featuresClone := c.Features.Copy()
+	if featuresClone != nil {
+		clone.Features = *featuresClone
 	}
+
```

Comment:
Why was this working for the global case but not for teams?

Reply:
because `AppConfig` implements a clone function that does a `clone = *c`

Reply:
that's correct! sorry for not making that clear before.